pymc/gp/gp.py

```diff
@@ -37,6 +37,26 @@
 __all__ = ["Latent", "Marginal", "TP", "MarginalApprox", "LatentKron", "MarginalKron"]
 
 
+_noise_deprecation_warning = (
+    "The 'noise' parameter has been been changed to 'sigma' "
+    "in order to standardize the GP API and will be "
+    "deprecated in future releases."
+)
+
+
+def _handle_sigma_noise_parameters(sigma, noise):
+    """Helper function for transition of 'noise' parameter to be named 'sigma'."""
+
+    if (sigma is None and noise is None) or (sigma is not None and noise is not None):
+        raise ValueError("'sigma' argument must be specified.")
```

Comment:
Maybe, the logic here could be:
```
if `sigma is not None`: good
else:
      if `noise is not None`: raise warning
      else: raise error.
```

OR
```
if `sigma is None`:
   if `noise is None`: raise error
   else: raise warning
```

Reply:
If both `sigma` and `noise` are specified, it is not really wrong, right? :) The user just provides more inputs than what is needed.

Reply:
I hear what you are staying but thought it might be fine to nudge them toward the new input. Not likely for both to be used, but thought being explicit about the intended behavior would be best instead of throwing away or checking the two parameters silently.

Reply:
If the user passes in sigma=0.1 and noise=0.2, it's not really possible from the outside to know which value is going to be used in the model.  The user must be a bit confused then, and an error message would help them out.  Both your alternatives for doing the logic make sense to me too though @danhphan, but I think it also makes sense how it is.

Reply:
Well said @bwengals That's what I had in mind. 
Was trying to make it easier from the developer's point of view

Reply:
Hi, thanks @wd60622 and @bwengals for the clarification. Total agree with "nudge users toward the new sigma" :)